File: uebung_1/tast_2/prettyOut.cpp
#include <iostream>
using namespace std;
void ausgabe(char c);
void ausgabe(char *text);
void ausgabe(char *text, char randZeichen);

int main()
{
  ausgabe('a');
  ausgabe("hallo welt");
  ausgabe("hallo welt", '+');

  return 0;
}

void ausgabe(char c)
{
  cout << "***\n*" << c << "*\n***\n";
}

void ausgabe(char *text)
{
  int length = strlen(text);
  string border = "**";
  for (int i = 0; i < length; i++)
  {
    border += "*";
  }

  cout << border << "\n*" << text << "*\n"
       << border << "\n";
}

void ausgabe(char *text, char randZeichen)
{
  int length = strlen(text);
  string border = "";
  border = randZeichen;
  border += randZeichen;

  for (int i = 0; i < length; i++)
  {
    border += randZeichen;
  }

  cout << border << "\n"
       << randZeichen << text
       << randZeichen << "\n"
       << border << "\n";
}

File: exercise_1/task_2/PrettyOut.java
package exercise_1.task_2;

public class PrettyOut {
  public static void main(String[] args) {
    ausgabe('d');
    ausgabe("hallo welt");
    ausgabe("hallo welt", '+');
  }

  static void ausgabe(char c) {
    System.out.println("***\n*" + c + "*\n***");
  }

  static void ausgabe(String text) {
    String border = "**";
    for (int i = 0; i < text.length(); i++) {
      border += "*";
    }
    System.out.println(border + "\n*" + text + "*\n" + border);
  }

  static void ausgabe(String text, char randZeichen) {
    String border = "" + randZeichen + randZeichen;
    for (int i = 0; i < text.length(); i++) {
      border += randZeichen;
    }
    System.out.println(border + "\n" + randZeichen + text + randZeichen + "\n" + border);
  }
}